### ISSUE TITLE
Black produced different code on the second pass of the formatter.

### ISSUE THREAD

**Issue by ddodt** [psf/black#2462]
**Description**
error: cannot format /home/ddodt/rockestate_branch/api/utils.py: INTERNAL ERROR: Black produced different code on the second pass of the formatter.  Please report a bug on https://github.com/psf/black/issues.  This diff might be helpful: /tmp/blk_bp4imlkr.log

**To Reproduce**
I am assuming that the log will help:
```
Mode(target_versions=set(), line_length=88, string_normalization=True, experimental_string_processing=False, is_pyi=False)
--- source
+++ first pass
@@ -340,14 +340,18 @@
 def flood_to_html():
     return table_style + pd.read_csv(
         os.path.join(module_path, "docs", "flood.csv"), sep=";", dtype={"Level": str}
     ).fillna("").to_html(index=False)
 
+
 def flood_computation_to_html():
     return table_style + pd.read_csv(
-        os.path.join(module_path, "docs", "flood_computation.csv"), sep=";", dtype={"Level": str}
+        os.path.join(module_path, "docs", "flood_computation.csv"),
+        sep=";",
+        dtype={"Level": str},
     ).fillna("").to_html(index=False)
+
 
 import subprocess
 from os_tools import TempDir
 
 
--- first pass
+++ second pass
@@ -342,15 +342,20 @@
         os.path.join(module_path, "docs", "flood.csv"), sep=";", dtype={"Level": str}
     ).fillna("").to_html(index=False)
 
 
 def flood_computation_to_html():
-    return table_style + pd.read_csv(
-        os.path.join(module_path, "docs", "flood_computation.csv"),
-        sep=";",
-        dtype={"Level": str},
-    ).fillna("").to_html(index=False)
+    return (
+        table_style
+        + pd.read_csv(
+            os.path.join(module_path, "docs", "flood_computation.csv"),
+            sep=";",
+            dtype={"Level": str},
+        )
+        .fillna("")
+        .to_html(index=False)
+    )
 
 
 import subprocess
 from os_tools import TempDir
```

**Expected behavior**

:) unambiguous formatting!

**Environment (please complete the following information):**

- Version:20.8b1
- OS and Python version: Linux/Python 3.8.8 | packaged by conda-forge | (default, Feb 20 2021, 16:22:27) 
[GCC 9.3.0]

**Does this bug also happen on main?**

not with <https://black.vercel.app/?version=main>

**Additional context**

**Comment by JelleZijlstra** [comment]
Thanks for the report, but since this is fixed on main, there's nothing for us to do. Please upgrade your Black version so you don't run into this crash.

**Comment by ichard26** [comment]
For more context, the problem here has to do with the "magic trailing feature" where a trailing comma causes Black to explode the container. Unfornately since Black can also add trailing commas, this quickly became unstable with the first pass adding a trailing comma and then the second pass coming along going "oh hey look at this trailing comma let's explode this!" even though it wasn't the user who added that comma. We eventually fixed this with an additional format pass (shouldn't affect performance in most cases since a second pass was already required for safety purposes). This fix is only available in 21.4b0 and higher. Here's the output with this version:

```diff
--- test.py	2021-08-30 17:04:21.775691 +0000
+++ test.py	2021-08-30 17:10:57.179657 +0000
@@ -1,4 +1,11 @@
 def flood_computation_to_html():
-    return table_style + pd.read_csv(
-        os.path.join(module_path, "docs", "flood_computation.csv"), sep=";", dtype={"Level": str}
-    ).fillna("").to_html(index=False)
+    return (
+        table_style
+        + pd.read_csv(
+            os.path.join(module_path, "docs", "flood_computation.csv"),
+            sep=";",
+            dtype={"Level": str},
+        )
+        .fillna("")
+        .to_html(index=False)
+    )
```

**Comment by ddodt** [comment]
Thanks! I worked around the issue by make a few more (unrelated I would have thought) modifications to the code. 
Humble suggestion: Maybe make reproducing the bug on  https://black.vercel.app/?version=main the first step of the bug report :)

Thanks for all the work, black is a great tool for us!
